### PR TITLE
feat: get clusterPort

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

Closes eggjs/egg#219

### CODE REVIEW

File: lib/master.js [hunk 1]
```diff
@@ -77,7 +78,16 @@ class Master extends EventEmitter {
 
     this.onExit();
 
-    this.forkAgentWorker();
+    getFreePort((err, port) => {
```
Comment (popomore) [inline lib/master.js]: 把这个方法放 util 里吧

Comment (atian25) [inline lib/master.js]: 嗯, 可以抽出来, 在 egg-bin 那边也用到了

https://github.com/eggjs/egg-bin/blob/master/lib/helper.js#L76

Comment (atian25) [inline lib/master.js]: https://github.com/node-modules/detect-port/pull/13

**Comment by popomore** [comment]
@fengmk2 @gxcsoccer 这个看看

File: lib/master.js [hunk 2]
```diff
@@ -77,7 +78,17 @@ class Master extends EventEmitter {
 
     this.onExit();
 
-    this.forkAgentWorker();
+    detectPort((err, port) => {
```
Comment (atian25) [inline lib/master.js]: 这个加进去后, egg-bin 那边还需要检测么?

Comment (popomore) [inline lib/master.js]: egg-bin 是另外一个功能？这个是 cluster-client 的端口

Comment (atian25) [inline lib/master.js]: 喔喔

**Comment by popomore** [comment]
发个 minor

**Comment by popomore** [comment]
+ egg-cluster@1.3.0

**Pull request merged**